src/bundle.c

```diff
@@ -530,17 +582,39 @@ static gboolean convert_images(RaucManifest *manifest, const gchar *dir, const g
 		if (!image->convert)
 			continue;
 
+		g_autofree gchar *tar_extracted = NULL;
+		g_autofree gchar *tar_extracted_path = NULL;
+		/* extract tar early if we need it for other outputs */
+		if (g_strv_contains((const gchar * const *)image->convert, "tar-extract") ||
+		    g_strv_contains((const gchar * const *)image->convert, "composefs")) {
+			g_debug("extracting tar artifact image '%s'", image->filename);
+			tar_extracted = convert_tar_extract(image, dir, fakeroot, &ierror);
+			if (!tar_extracted) {
+				g_propagate_error(error, ierror);
+				return FALSE;
+			}
+			tar_extracted_path = g_build_filename(dir, tar_extracted, NULL);
+		}
+
 		gboolean keep = FALSE;
 		g_autoptr(GPtrArray) converted = g_ptr_array_new_with_free_func(g_free);
 		for (gchar **method = image->convert; *method != NULL; method++) {
 			gchar *converted_filename = NULL;
 			if (g_str_equal(*method, "tar-extract")) {
-				g_debug("extracting tar artifact image '%s'", image->filename);
-				converted_filename = convert_tar_extract(image, dir, fakeroot, &ierror);
+				converted_filename = g_strdup(tar_extracted);
+			} else if (g_str_equal(*method, "composefs")) {
```

Comment:
When RAUC is compiled without composefs support, this conversion method will fail. I've made this explicit now.